### ISSUE TITLE
Big slowdown when loading lot of data in viewmodel

### ISSUE THREAD

**Issue by GiampaoloGabba** [OYIon/LiveSharp#52]
Hello,
i have a ViewModel that read a bunch of data from a Json file (7k records with some properties) 

The reading is done async when the viewmodel load (not in constructor but when the viewmodel is fully loaded: OnNavigatedTo event from Prism)

Without Livesharp everything is fine: the list will load in under 1 sec (both phone and emulator), when i use LiveSharp everything get stuck for 10 sec in emulator and 3-4 on phone.

Plus, everytime i close and reopen the same page the time will double:
- 20 seconds the second time i open the page
- 30 seconds the third time, ecc...

I tried to clear the list when closing the viewmodel but without success

**Comment by ionoy** [comment]
Thanks for the feedback!

Can you show me the code loading the records?

**Comment by GiampaoloGabba** [comment]
```
public override void OnNavigatedTo(INavigationParameters parameters)
{
	Task.Run(PopolaFarmacie);
	base.OnNavigatedTo(parameters);
}
```

```
private async Task PopolaFarmacie()
{
        var listaFamarmacie = JsonDataReader.Read<List<Farmacia>>("farmacie.json"); // <---  Slowdown here
        
        ListaFarmacie  = listaFamarmacie

}
```

Where ListaFarmacia is a IENumerable binded to the view.

```
public static class JsonDataReader
{
	public static T Read<T>(string jsonFileName)
	{
		T serialized;
		var assembly  = typeof(App).GetTypeInfo().Assembly;
		Stream stream = assembly.GetManifestResourceStream($"{assembly.GetName().Name}.Resources.{jsonFileName}");

		if (stream == null)	return default;

		using (var reader = new StreamReader(stream))
		{
			var jsonString   = reader.ReadToEnd();
			serialized = JsonConvert.DeserializeObject<T>(jsonString);
		}

		return serialized;
	}
}
```
The json is about 7k records of this:

```
public class Farmacia
{
	public string Indirizzo { get; set; }
	public string Nome { get; set; }
	public double Lat { get; set; }
	public double Longi { get; set; }
}
```

I also tried to load the same data from a Sqlite DB using sqlite-net, and even translated the json in a c# class but nothing change. Everytime i load the data with LiveSharp i have to wait an incremental time that increases every time i close and open the page in the same session.

Btw i'm using Fody & Prism.

**Comment by ionoy** [comment]
Does it still happen even if you don't change any code?

**Comment by GiampaoloGabba** [comment]
yepp! i just start a debug session and visit the page without changing nothing. Then i pop the page and push it another time (using a navigation page, inside a tab) whitout touching nothing

**Comment by ionoy** [comment]
Do you have `INotifyPropertyChanged` injected in your models? 

This code: https://github.com/OYIon/LiveSharp/blob/master/src/LiveSharp.Support.XamarinForms/XamarinFormsUpdateHandler.cs#L27 along with `GetUpdate` is the only thing that gets called even if you haven't updated anything.

**Comment by GiampaoloGabba** [comment]
Yes, i'm using [Fody.PropertyChanged](https://github.com/Fody/PropertyChanged).
The property `ListaFarmacie` is the one that gets injected with `INotifyPropertyChanged `

**Comment by ionoy** [comment]
Please try with this update.
[LiveSharp.1.1.34.zip](https://github.com/OYIon/LiveSharp/files/4235302/LiveSharp.1.1.34.zip)
